python/lsst/daf/butler/tests/_testRepo.py

```diff
@@ -106,7 +104,7 @@ def makeTestCollection(repo):
     """
     # Create a "random" collection name
     # Speed matters more than cryptographic guarantees
-    collection = "test" + "".join((str(i) for i in np.random.randint(0, 10, size=8)))
+    collection = "test" + "".join((str(i) for i in Random().sample(range(10), 8)))
```

Comment:
I'm not sure it's a good idea to construct a new `Random` every time, since that might introduce weird correlations (e.g., slowly varying seeds). I suggest storing a persistent `Random` in the module, or (I think equivalently?) using the `sample` function.

Reply:
`sample` seems like a very roundabout way of doing this (as does the original method, to be honest). I think this would be much simpler with:
```suggestion
    collection = "test" + str(random.randrange(1_000_000_000))
```

Reply:
I was also pondering this:
```
>>> tempfile.mktemp(dir="", prefix="test")
'testzp6hm_z8'
```
which seems good enough.

Reply:
That would work, but it's deprecated since Python 2.3... 😦